File: generateData/sendDataTokafka/src/main/java/org/example/config/SparkConfig.java
package org.example.config;

import org.apache.spark.sql.SparkSession;

public class SparkConfig {
    private final SparkSession sparkSession;

    public SparkConfig() {
        this.sparkSession = SparkSession.builder()
                .appName("Send Json to Kafka")
                .master("local[*]")
                .getOrCreate();
    }

    public SparkSession getSparkSession() {
        return sparkSession;
    }

    public void stopSpark() {
        if (sparkSession != null) {
            sparkSession.stop();
        }
    }
}

File: batch/silver/src/main/scala/config/SparkConfig.scala
package config

import org.apache.spark.sql.SparkSession

object SparkConfig {
  def createSession(): SparkSession = {
    SparkSession.builder()
      .appName("Kafka to Delta Table")
      .master("local[*]")
      .config("spark.sql.extensions", "io.delta.sql.DeltaSparkSessionExtension")
      .config("spark.sql.catalog.spark_catalog", "org.apache.spark.sql.delta.catalog.DeltaCatalog")
      .getOrCreate()
  }
}